src/library/Uno.Cupertino/Styles/Controls/ToggleSwitch.xaml

```diff
@@ -0,0 +1,382 @@
+							<ColumnDefinition Width="*" />
+						</Grid.ColumnDefinitions>
+
+						<TextBlock Text="{TemplateBinding Header}"
+								   Style="{StaticResource CupertinoBaseTextBlockStyle}"
+								   Foreground="{StaticResource CupertinoLabelBrush}"
+								   Visibility="{Binding Header, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource MaterialEmptyToCollapsed},TargetNullValue=Collapsed}"
+								   Grid.ColumnSpan="4" />
+
+						<ContentPresenter x:Name="OnContentPresenter"
+										  Grid.Row="1"
+										  Grid.RowSpan="3"
+										  Grid.Column="2"
+										  Opacity="0"
+										  Foreground="{TemplateBinding Foreground}"
+										  IsHitTestVisible="False"
+										  HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
+										  VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
+										  AutomationProperties.AccessibilityView="Raw" />
+
+						<Grid Grid.Row="1"
+							  Grid.RowSpan="3"
+							  Grid.ColumnSpan="3"
+							  Control.IsTemplateFocusTarget="True" />
+
+						<!-- Marker element used to compute draggable zone -->
+						<Rectangle x:Name="SwitchKnobBounds"
+								   Grid.Row="2" />
+
+						<Grid  x:Name="ToggleBackground"
+							   Grid.Row="2"
+							   VerticalAlignment="Center">
+							<Rectangle x:Name="OuterBorder"
+									   Height="31"
+									   Width="51"
+									   RadiusX="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource TopLeftCornerRadiusDoubleValueConverter}}"
+									   RadiusY="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource BottomRightCornerRadiusDoubleValueConverter}}"
+									   Fill="{StaticResource MaterialToggleSwitchOffBackgroundBrush}" />
+
+							<Rectangle x:Name="HighlightBorder"
+									   Height="31"
+									   Width="51"
+									   RadiusX="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource TopLeftCornerRadiusDoubleValueConverter}}"
+									   RadiusY="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource BottomRightCornerRadiusDoubleValueConverter}}"
+									   Fill="{TemplateBinding Background}"
+									   Opacity="0" />
+						</Grid>
+
+						<Grid x:Name="SwitchKnob"
+							  Grid.Row="2"
+							  Padding="2,0"
+							  HorizontalAlignment="Left"
+							  Width="31"
+							  Height="27">
+
+							<Grid CornerRadius="60">
+								<toolkit:ElevatedView x:Name="SwitchKnobOn"
+													  Elevation="3"
+													  VerticalAlignment="Stretch"
+													  HorizontalAlignment="Stretch"
+													  Background="Transparent"
+													  Opacity="0">
+									<Ellipse Fill="{TemplateBinding Foreground}"
+											 VerticalAlignment="Stretch"
+											 HorizontalAlignment="Stretch" />
+								</toolkit:ElevatedView>
+							</Grid>
+
+							<Grid CornerRadius="60">
```

Comment:
Setting CornerRadius directly on the ElevatedView didn't work?

Reply:
It was giving a weird style on Android. Seems to be an Uno issue. Might be related to this https://github.com/unoplatform/nventive-private/issues/176#issuecomment-788083386